File: GestionRessourcesInfo/src/main/java/com/example/GestionRessourcesInfo/model/Equipement.java
package com.example.GestionRessourcesInfo.model;

import com.example.GestionRessourcesInfo.model.enums.EtatEquipement;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Equipement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String description;
    @Enumerated(EnumType.STRING)
    private EtatEquipement etat;
    private Date dateAcquisition;

    @OneToMany(mappedBy = "equipement")
    private List<Panne> pannes;

}

File: FrontEnd/front-endSi/src/app/models/equipement.ts
import { EtatEquipement } from './enums/etat-equipement';
import { Panne } from './panne';

export interface Equipement {
  id?: number;
  nom: string;
  description: string;
  etat: EtatEquipement;
  dateAcquisition: string;
  pannes?: Panne[];
}


